gulpfile.js

```diff
@@ -85,6 +86,84 @@ function createReloadServer() {
   });
 }
 
+function writeServiceWorkerFile() {
+  let path = require('path');
+  let rootDir = 'dist';
+
+  let dynamicUrlToDependencies = {};
+  let partialTemplateFiles = ['head-meta.html', 'site-nav.html']
+    .map(file => path.join(rootDir, 'templates', file));
+
+  ['about', 'index', 'shell'].forEach(htmlFile => {
+    let dynamicUrl = '/' + htmlFile;
+    let dependencies = partialTemplateFiles.concat(`${rootDir}/${htmlFile}.html`);
+    dynamicUrlToDependencies[dynamicUrl] = dependencies;
+    // Treat / like /index
+    if (htmlFile === 'index') {
+      dynamicUrlToDependencies['/'] = dependencies;
+    }
+  });
+
+  let config = {
+    cacheId: 'polymerproject',
+    staticFileGlobs: [
+      `${rootDir}/images/logos/p-logo.png`,
+      `${rootDir}/elements/**`,
+      `${rootDir}/js/*.js`,
+      `${rootDir}/css/*.css`,
+      `${rootDir}/bower_components/**/webcomponents-lite.min.js`,
+      `${rootDir}/bower_components/highlight/highlight.js`,
+    ],
+    dynamicUrlToDependencies: dynamicUrlToDependencies,
+    navigateFallback: '/shell',
```

Comment:
Is there a way to specify our own custom handler in `runtimeCaching` config and remove `navigateFallback` to get this desired behavior? Also, is this something that sw-helpers will be able to handle? Seems like a pretty good use case for non-shell type (i.e. server-generated) sites (i.e. show me the page if it has been runtime cached, or show me a fallback if it hasn't).